charts/generic-prometheus-alerts/ci/tests/ingress-alerts-tests.yaml

```diff
@@ -0,0 +1,170 @@
+---
+rule_files:
+  - ../ingress-alerts.yaml
+
+evaluation_interval: 1m
+
+tests:
+  - interval: 1m
+    input_series:
+      - series: 'nginx_ingress_controller_requests{exported_namespace="test-application-dev", ingress="test-application-ingress", status="500"}'
+        values: '0 1 3'
+      - series: 'nginx_ingress_controller_requests{exported_namespace="test-application-dev", ingress="test-application-ingress", status="501"}'
+        values: '0 2 5'
+
+    name: 5xxErrorResponses alert fired as request rate increases over the duration of the 1 minute time range window
+
+    promql_expr_test:
+      # assert that
+      #   at 2m the rate (increase per second) in the last minute: for 500 requests it's 0.03 ((3 - 1) / 60), for 501 requests it's 0.05 ((5 - 2) / 60)
+      #   which multiplied by 60 is greater than 0
+      #   the average of those is 2.5
+      - expr: avg(rate(nginx_ingress_controller_requests{exported_namespace="test-application-dev", ingress="test-application-ingress", status=~"5.*"}[1m]) * 60 > 0) by (ingress, exported_namespace)
+        eval_time: 2m
+        exp_samples:
+          - labels: '{exported_namespace="test-application-dev", ingress="test-application-ingress"}'
+            value: 2.5
+
+    alert_rule_test:
+      # assert that the 5xx alert is firing
+      - eval_time: 2m
+        alertname: 5xxErrorResponses
+        exp_alerts:
+          - exp_labels:
+              alertname: 5xxErrorResponses
+              exported_namespace: test-application-dev
+              ingress: test-application-ingress
+              severity: alert-severity-tag
+            exp_annotations:
+              message: "Ingress test-application-dev/test-application-ingress is serving 5xx responses."
+              runbook_url: https://runbook.com/ingress-5xx-error-responses
+              dashboard_url: https://grafana.live.cloud-platform.service.justice.gov.uk/d/golden-signals/golden-signals?orgId=1&var-namespace=test-application-dev&var-service=test-application
+
+  - interval: 1m
+    input_series:
+      - series: 'nginx_ingress_controller_requests{exported_namespace="test-application-dev", ingress="test-application-ingress", status="500"}'
+        values: '0 1 0'
+
+    name: 5xxErrorResponses alert not fired as request rate decreases over the duration of the 1 minute time range window
+
+    promql_expr_test:
+      # at 2m the rate (increase per second) in the last minute: 0 and multiplying that by 60 is still 0 which is not greater than 0, assert nothing is returned
+      - expr: avg(rate(nginx_ingress_controller_requests{exported_namespace="test-application-dev", ingress="test-application-ingress", status=~"5.*"}[1m]) * 60 > 0) by (ingress, exported_namespace)
```

Comment:
Am I right in thinking that if we are checking `> 0` then multiplying by 60 doesn't really achieve anything? Given the confusion it causes maybe we could not bother multiplying?

Reply:
That's a good point. `rate` is the increase _per second_ for a counter and we multiply it by 60 to get the number _per minute_. I assumed we multiplied everywhere because in `ModSecurityBlocking` this is included the alert message
> `Blocking http requests at rate of {{`{{`}} printf "%0.0f" $value {{`}}`}} per minute`

We could drop the multiplication from all the expressions and update the comment. What do you think?

Reply:
Every time I see that `[1m]) * 60` my brain starts jumping to conclusions that don't make sense. 1m and 60 appear to be related but it's just an unhappy coincidence. So yes I'd like to remove it and update the comment.